### ISSUE TITLE
Very high CPU usage after steamwebhelper segfaults

### ISSUE THREAD

**Issue by guihkx** [ValveSoftware/steam-for-linux#5061]
#### Your system information

https://gist.githubusercontent.com/GUiHKX/537aa3addda6f0eb3184c3414dde1374/raw/6444f830ad8b3fcd64bd8aa6cea0cccc6809f7fb/gistfile1.txt

* Steam client version (build number or date): 1499120811
* Distribution (e.g. Ubuntu): Arch Linux 
* Opted into Steam client beta?: YES
* Have you checked for system updates?: YES

#### Please describe your issue in as much detail as possible:
I can't reproduce the "steamwebhelper" crash, but I was simply browsing through the Steam Store and nothing really showed up on screen (no errors or something). Then I checked `dmesg` and saw this:

```
[24679.485725] steamwebhelper[23492]: segfault at 79566c ip 000055f774aa6062 sp 00007ffe24aa4ce8 error 6 in steamwebhelper[55f774967000+2f5000]
```

Here's a Htop screenshot:

![screenshot_2017-07-04_23-40-35](https://user-images.githubusercontent.com/626206/27847138-69f1fcd2-6112-11e7-9287-777f49271f3a.png)

**Comment by ericek111** [comment]
I'm having exactly the same issue. Does it really take more than 4 days to fix it?

**System Information:** https://gist.github.com/ericek111/7db631690dffe8b1495a5683dc051e30
Steam build:`1499366534`
Ubuntu MATE 16.04, Linux 4.4.0-83

![image](https://user-images.githubusercontent.com/7243565/27992663-e9b9b646-6499-11e7-81e4-cafe5f0c0427.png)

**Comment by kisak-valve** [comment]
Hello @GUiHKX, @ericek111, please opt into the steam beta client and see if you can reproduce this issue.

**Comment by GUiHKX** [comment]
I've just checked for updates but the Steam client says it is already updated (I'm still in the beta branch)...

**Comment by kisak-valve** [comment]
I did reproduce this issue with the 2017-07-11 steam beta client, but it should be harder to cause the steamwebhelper crash leading up to it.

If opting-out of the steam beta, restarting steam, then switching back to the beta does not bring it up to date, then check if there is a steam.cfg file in steam's folders (from https://www.reddit.com/r/Steam/comments/5vuwfz/i_cant_convince_steam_to_optin_to_beta/).

**Comment by GUiHKX** [comment]
I had to `SIGKILL` the steam process and open it again, then the auto-update worked (I did this before you posted the steam.cfg thing, but thank you anyway). Like you said, this update didn't fix it, and I actually found a way to reproduce it 100%:

1. Open CS:GO and wait for the blog page to load up.
2. Switch to the "WATCH" tab.
3. Go back to "HOME".
4. The blog page won't load up anymore because `steamwebhelper` `SEGFAULT`ed.
5. However, if you repeat these steps, `steamwebhelper` will start to work again (but the CPU usage will still be stuck at 100%).

This is really bad for me because I mostly play CS:GO, so having this high CPU usage will worse my FPS. :confused: 

Here's a video of all of that:

https://youtu.be/MQHK_UCUTlY

**Comment by tpruzina** [comment]
Can confirm this:
```
...
Timed out waiting for mutex in PutInternal()
Timed out waiting for mutex in PutInternal()
chrome_ipc_client.cpp (792) : Assertion Failed: Failure pushing command to steamwebhelper.exe
Assert( Assertion Failed: Failure pushing command to steamwebhelper.exe ):chrome_ipc_client.cpp:792

Installing breakpad exception handler for appid(steam)/version(1499909773)
assert_20170714000332_22.dmp[20414]: Uploading dump (out-of-process)
/tmp/dumps/assert_20170714000332_22.dmp
Timed out waiting for mutex in PutInternal()
assert_20170714000332_22.dmp[20414]: Finished uploading minidump (out-of-process): success = yes
assert_20170714000332_22.dmp[20414]: response: CrashID=bp-5988cb3e-0c75-4980-931a-e69c42170713
assert_20170714000332_22.dmp[20414]: file ''/tmp/dumps/assert_20170714000332_22.dmp'', upload yes: ''CrashID=bp-5988cb3e-0c75-4980-931a-e69c42170713''
Timed out waiting for mutex in PutInternal()
Timed out waiting for mutex in PutInternal()
Timed out waiting for mutex in PutInternal()
Timed out waiting for mutex in PutInternal()
Timed out waiting for mutex in PutInternal()
Timed out waiting for mutex in PutInternal()
Timed out waiting for mutex in PutInternal()
Timed out waiting for mutex in PutInternal()
Timed out waiting for mutex in PutInternal()
Timed out waiting for mutex in PutInternal()
[S_API FAIL] SteamAPI_Init() failed; create pipe failed.Timed out waiting for mutex in PutInternal()
Timed out waiting for mutex in GetInternal()
Timed out waiting for mutex in GetInternal()
../common/html/chrome_ipc_server.cpp (459) : Assertion Failed: Failure reading client command mem stream
../common/html/chrome_ipc_server.cpp (441) : Assertion Failed: Got over 32MB message in CEFThread Command Stream!
Timed out waiting for mutex in PutInternal()
Timed out waiting for mutex in PutInternal()
Timed out waiting for mutex in PutInternal()
Timed out waiting for mutex in PutInternal()
^C
zsh: interrupt  steam
```
Not sure when this started, but steam UI is basically unusable after the bug is triggered.

**Comment by steamlinux** [comment]
I run gmod
gmod crashes
after launch I have this in dmesg
```
[160532.053368] steamwebhelper[22980]: segfault at 2c0778405410 ip 00007f2502dd71dc sp 00007ffd87afad60 error 4 in libjemalloc.so.1[7f2502dd3000+34000]
[166053.999332] steamwebhelper[23106]: segfault at 1e3344805968 ip 00007fd7580471dc sp 00007ffe0907f7e0 error 4 in libjemalloc.so.1[7fd758043000+34000]
[166077.340528] CompositorTileW[27758]: segfault at 2df92d805f38 ip 00007f4aacf171dc sp 00007f4a923fa470 error 4 in libjemalloc.so.1[7f4aacf13000+34000]
[166077.348277] steamwebhelper[27628]: segfault at 7f9589757f20 ip 00007f9524fff3ad sp 00007fff4ca36930 error 6 in libjemalloc.so.1[7f9524ffb000+34000]
[166084.663308] CompositorTileW[27852]: segfault at 7f266d8a4bc0 ip 00007f237fdcf3ad sp 00007f2364ff8fd0 error 6 in libjemalloc.so.1[7f237fdcb000+34000]
[166084.663644] CompositorTileW[27853]: segfault at 7f2674488e40 ip 00007f237fdcf3ad sp 00007f23647f7fd0 error 6 in libjemalloc.so.1[7f237fdcb000+34000]
[166084.671399] steamwebhelper[27775]: segfault at ffffe091d5e27385 ip 00007f9baf3fcb20 sp 00007fff6185b0d8 error 5 in libpthread-2.24.so[7f9baf3f3000+18000]
[166091.745270] CompositorTileW[27964]: segfault at 19368c4046f0 ip 00007fe5128c71dc sp 00007fe4f6bf6fd0 error 4 in libjemalloc.so.1[7fe5128c3000+34000]
[166091.750885] steamwebhelper[27892]: segfault at 7f13fd4362a8 ip 00007f124e68f3ad sp 00007fff054fc680 error 6 in libjemalloc.so.1[7f124e68b000+34000]
```

**Comment by tpruzina** [comment]
Temporary workaround is preventing steamwebhelper from running until this gets fixed.

```chmod -x ~/.local/share/Steam/ubuntu12_64/steamwebhelper```

**Comment by marc2377** [comment]
I'm having this as well.

`[  987.742892] steamwebhelper[3965]: segfault at 84ba0 ip 000055f09a764f82 sp 00007ffd62689a38 error 6 in steamwebhelper[55f09a626000+2f5000]`

System information:
https://gist.github.com/marc2377/18f12f5609af38da031e92f42337b70f/

I was previously using the amdgpu driver, and switched to the radeon driver in an attempt to fix overlay screenshots (f12 key) not working on HL2 Episode 2. I'm creating an issue about that. Anyway, both problems persist across drivers.

**Comment by tpruzina** [comment]
Hm, anyone here running a firewall? I noticed that after flushing my iptables rules the problems went away. I also tried reverting back to "simple stateful firewall" rules such as these:

> :INPUT DROP
> :OUTPUT ACCEPT
> -A INPUT -m state --state RELATED,ESTABLISHED -j ACCEPT

And the thing still breaks. Haven't used steam in a while to say when my problem appeared or even if it's the same thing you guys have problem with, could be a separate bug.

**Comment by steamlinux** [comment]
after segfault and high load cpu, I kill process "steamwebhelper.sh"
steam doesn't crash, I don't have high load cpu after this
I killed proccess ~20 hours ago
small solution?

**Comment by tpruzina** [comment]
You could just ```chmod -x .local/share/Steam/ubuntu12_64/steamwebhelper``` I suppose.
Webstore won't work, but I suppose at least Valve devs have some motivation to fix that *** up for once.

**Comment by GUiHKX** [comment]
I cannot reproduce the crash anymore using [this previous method](https://github.com/ValveSoftware/steam-for-linux/issues/5061#issuecomment-314615581) on the latest beta update...

**Comment by kisak-valve** [comment]
Per "Fixed an unintentional issue where Steam clients could experience sluggish frame rate if the Steam Web Helper process could not connect to our error- and crash-reporting servers." in the [2017-07-25](https://steamcommunity.com/groups/SteamClientBeta/announcements/detail/1447197096908335573) steam beta client, please opt into the beta client and retest this issue.

**Comment by GUiHKX** [comment]
July 17's update actually fixed it for me. I've checked the logs:

```
$ journalctl --since "2017-07-17" | grep steamwebhelper
```

It produces no output.

**Comment by steamlinux** [comment]
I have this bug now, with beta updates 2017-07-25
```
# journalctl --since "2017-07-17" | grep steamwebhelper
июл 25 21:02:23 ruslan kernel: steamwebhelper[7049]: segfault at 7f594f424330 ip 00007f5201eef3ad sp 00007ffc965205e0 error 6 in libjemalloc.so.1[7f5201eeb000+34000]
июл 25 21:03:36 ruslan kernel: steamwebhelper[8986]: segfault at 7f1f4e910150 ip 00007f1d5759f3ad sp 00007ffe57892f00 error 6 in libjemalloc.so.1[7f1d5759b000+34000]
июл 25 21:14:30 ruslan kernel: steamwebhelper[9433]: segfault at 7f9939d093c0 ip 00007f9838acf3ad sp 00007fff50d203e0 error 6 in libjemalloc.so.1[7f9838acb000+34000]
июл 25 21:46:37 ruslan kernel: steamwebhelper[12904]: segfault at 7f9703502950 ip 00007f97032cf20d sp 00007ffe16febe10 error 4 in libjemalloc.so.1[7f97032cb000+34000]
июл 26 01:20:32 ruslan kernel: steamwebhelper[23597]: segfault at 3e ip 00007f9904b94b20 sp 00007ffe14fd3058 error 4 in libpthread-2.24.so[7f9904b8b000+18000]
июл 26 01:20:40 ruslan kernel: steamwebhelper[21185]: segfault at 11e014e19 ip 00007f77d4becb20 sp 00007fffb8368628 error 4 in libpthread-2.24.so[7f77d4be3000+18000]
июл 26 09:00:07 ruslan kernel: steamwebhelper[21306]: segfault at 7f0a654aa940 ip 00007f07764773ad sp 00007ffd71a95de0 error 6 in libjemalloc.so.1[7f0776473000+34000]
июл 26 13:42:16 ruslan kernel: steamwebhelper[4860]: segfault at 7f4c4c968170 ip 00007f4a56fff3ad sp 00007ffe2d2b0980 error 6 in libjemalloc.so.1[7f4a56ffb000+34000]
июл 26 13:42:35 ruslan kernel: steamwebhelper[29936]: segfault at fffff1e0604daacc ip 00007f66495acb20 sp 00007ffdfb387678 error 5 in libpthread-2.24.so[7f66495a3000+18000]
июл 26 14:40:20 ruslan kernel: steamwebhelper[30081]: segfault at 7f54454bdd00 ip 00007f4ec81273ad sp 00007fffd67b6ab0 error 6 in libjemalloc.so.1[7f4ec8123000+34000]
июл 26 14:40:25 ruslan kernel: steamwebhelper[16243]: segfault at 7faba14ba470 ip 00007faba128720d sp 00007fff44422b60 error 4 in libjemalloc.so.1[7faba1283000+34000]
```
```
# journalctl --since "2017-07-25" | grep steamwebhelper
июл 25 21:02:23 ruslan kernel: steamwebhelper[7049]: segfault at 7f594f424330 ip 00007f5201eef3ad sp 00007ffc965205e0 error 6 in libjemalloc.so.1[7f5201eeb000+34000]
июл 25 21:03:36 ruslan kernel: steamwebhelper[8986]: segfault at 7f1f4e910150 ip 00007f1d5759f3ad sp 00007ffe57892f00 error 6 in libjemalloc.so.1[7f1d5759b000+34000]
июл 25 21:14:30 ruslan kernel: steamwebhelper[9433]: segfault at 7f9939d093c0 ip 00007f9838acf3ad sp 00007fff50d203e0 error 6 in libjemalloc.so.1[7f9838acb000+34000]
июл 25 21:46:37 ruslan kernel: steamwebhelper[12904]: segfault at 7f9703502950 ip 00007f97032cf20d sp 00007ffe16febe10 error 4 in libjemalloc.so.1[7f97032cb000+34000]
июл 26 01:20:32 ruslan kernel: steamwebhelper[23597]: segfault at 3e ip 00007f9904b94b20 sp 00007ffe14fd3058 error 4 in libpthread-2.24.so[7f9904b8b000+18000]
июл 26 01:20:40 ruslan kernel: steamwebhelper[21185]: segfault at 11e014e19 ip 00007f77d4becb20 sp 00007fffb8368628 error 4 in libpthread-2.24.so[7f77d4be3000+18000]
июл 26 09:00:07 ruslan kernel: steamwebhelper[21306]: segfault at 7f0a654aa940 ip 00007f07764773ad sp 00007ffd71a95de0 error 6 in libjemalloc.so.1[7f0776473000+34000]
июл 26 13:42:16 ruslan kernel: steamwebhelper[4860]: segfault at 7f4c4c968170 ip 00007f4a56fff3ad sp 00007ffe2d2b0980 error 6 in libjemalloc.so.1[7f4a56ffb000+34000]
июл 26 13:42:35 ruslan kernel: steamwebhelper[29936]: segfault at fffff1e0604daacc ip 00007f66495acb20 sp 00007ffdfb387678 error 5 in libpthread-2.24.so[7f66495a3000+18000]
июл 26 14:40:20 ruslan kernel: steamwebhelper[30081]: segfault at 7f54454bdd00 ip 00007f4ec81273ad sp 00007fffd67b6ab0 error 6 in libjemalloc.so.1[7f4ec8123000+34000]
июл 26 14:40:25 ruslan kernel: steamwebhelper[16243]: segfault at 7faba14ba470 ip 00007faba128720d sp 00007fff44422b60 error 4 in libjemalloc.so.1[7faba1283000+34000]
```
```
# dmesg
...
[55227.152073] CompositorTileW[25616]: segfault at 0 ip 00007f7af75773ad sp 00007f7adcff9230 error 6 in libjemalloc.so.1[7f7af7573000+34000]
[55232.413865] steamwebhelper[21306]: segfault at 7f0a654aa940 ip 00007f07764773ad sp 00007ffd71a95de0 error 6 in libjemalloc.so.1[7f0776473000+34000]
[56180.362599] usb 2-2: USB disconnect, device number 3
[56188.319315] usb 2-1: USB disconnect, device number 2
[72161.351623] steamwebhelper[4860]: segfault at 7f4c4c968170 ip 00007f4a56fff3ad sp 00007ffe2d2b0980 error 6 in libjemalloc.so.1[7f4a56ffb000+34000]
[72179.867729] steamwebhelper[29936]: segfault at fffff1e0604daacc ip 00007f66495acb20 sp 00007ffdfb387678 error 5 in libpthread-2.24.so[7f66495a3000+18000]
[75645.175881] steamwebhelper[30081]: segfault at 7f54454bdd00 ip 00007f4ec81273ad sp 00007fffd67b6ab0 error 6 in libjemalloc.so.1[7f4ec8123000+34000]
[75649.860345] steamwebhelper[16243]: segfault at 7faba14ba470 ip 00007faba128720d sp 00007fff44422b60 error 4 in libjemalloc.so.1[7faba1283000+34000]
[75658.116297] CompositorTileW[16418]: segfault at 18 ip 00007f854eb8cb20 sp 00007f853c7f9218 error 4 in libpthread-2.24.so[7f854eb83000+18000]
```

**Comment by steamlinux** [comment]
before updates 2017-07-25 I doesn't see "CompositorTileW" in dmesg

**Comment by Francesco149** [comment]
can confirm that this still happens on latest beta, except my crash is in libcef and I'm running steam inside a gentoo chroot

```
[77308.519603] traps: steamwebhelper[13064] trap int3 ip:7ff7c211fd7f sp:7ffffc0d8230 error:0
[77308.519606]  in libcef.so[7ff7bffc4000+75ff000]
[77308.521903] steamwebhelper[10408]: segfault at 0 ip 00007f57d4a22dc6 sp 00007ffd7f721b28 error 4 in libc-2.23.so[7f57d49a4000+190000]
[79372.522229] traps: steamwebhelper[17312] trap int3 ip:7fb05e1d84bf sp:7ffd8cf8e6a0 error:0
[79372.522231]  in libcef.so[7fb05c078000+760d000]
[79372.527148] steamwebhelper[16907]: segfault at 0 ip 00007f76174addc6 sp 00007fffab631128 error 4 in libc-2.23.so[7f761742f000+190000]
```

I had to chmod -x steamwebhelper to prevent constant cpu usage.

this bug has been around for years, it seems :(

**Comment by guihkx** [comment]
It's been over one year since I reported this and I haven't experienced any more crashes in a long time, so I'm closing this.

**Comment by huyouba1** [comment]
I have a similar problem. The CPU of the machine will soar around 3 a.m. every day, and similar exception logs will appear during this period

```
[8417590.289072] jhashcat[15881]: segfault at 0 ip 00007f2c7d834044 sp 00007ffd7544f7c0 error 4 in libc-2.17.so[7f2c7d7c6000+1c4000]
[8417593.618128] jhashcat[16012]: segfault at 0 ip 00007f0bc65cb044 sp 00007ffe2d1a7710 error 4 in libc-2.17.so[7f0bc655d000+1c4000]
[8417600.404113] jhashcat[16242]: segfault at 0 ip 00007f03ad2d9044 sp 00007ffe9d026900 error 4 in libc-2.17.so[7f03ad26b000+1c4000]
[8417613.813711] jhashcat[17278]: segfault at 0 ip 00007febf0cf1044 sp 00007fff31cb0060 error 4 in libc-2.17.so[7febf0c83000+1c4000]
[8417623.890128] jhashcat[17679]: segfault at 0 ip 00007f607c65a044 sp 00007fff99ab3470 error 4 in libc-2.17.so[7f607c5ec000+1c4000]
[8417637.300341] jhashcat[20801]: segfault at 0 ip 00007f3cfb534044 sp 00007fff7849a210 error 4 in libc-2.17.so[7f3cfb4c6000+1c4000]
[8417647.423766] jhashcat[21187]: segfault at 0 ip 00007f12a022b044 sp 00007ffe3a4e23f0 error 4 in libc-2.17.so[7f12a01bd000+1c4000]
[8417654.105707] jhashcat[21451]: segfault at 0 ip 00007f6fa91da044 sp 00007ffd78ae5bb0 error 4 in libc-2.17.so[7f6fa916c000+1c4000]
[8417660.828564] jhashcat[21656]: segfault at 0 ip 00007f0908fbd044 sp 00007ffeaaba5840 error 4 in libc-2.17.so[7f0908f4f000+1c4000]
```
Is there a solution now?

**Comment by BETLOG** [comment]
Earlier today I was attempting to use proton 8.0-101, GE-Proton8-2, and GE-Proton8-3 with Battlefield 4 (AppId=1238860).
At some point it seemed like the interface had been updated, despite me not recalling an update, and while steam was running.
A couple of times it seemed to reload steam entirely, but not the systemtray icon. Each time i had just deleted the compatdata folder of the old version and changed the proton version being forced in steam for bf4.

None of the protons I was trying worked either; the install and EA/DX/PB stuff happened fine up till the main BF4 window, which it only displayed locked-up and black.
If changing a games proton version in steam causes this 'reload everything' behaviour and is the new normal.... well, it feels very weird.
Numerous times the machine segfaulted and either rebooted or logged me out.

After a few hours I reverted to GE-Proton7-43 and hoped it wasn't going to be an ongoing problem.
Numerous reboots later (checking BIOS and misc settings, but not really altering anything) and i notice this in my dmesg output:

> steamwebhelper[2995]: segfault at 1001 ip 00007fdfa6109a92 sp 00007fff204e6140 error 4 in libcef.so[7fdfa24ef000+7770000] likely on CPU 3 (core 3, socket 0)

I'm not even sure I have started steam in this session, i dont think I have, and it is NOT set to auto start.

I can't tell if I should be expecting imminent hardware failure or if the most recent steam update has gone rogue.
